src/BatteryStateOfChargeService.h

```diff
@@ -19,4 +27,27 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
     const double initialStateOfChargePercent_;
 
     // Recommend adding necessary variables here
+    double initialCurrent;
+    double newCurrent = 0;
+    QTime initialTime;
+    QTime newTime;
+
+    double dTime;
+    double aveCurrent;
+    double dAmpHours;
+    double initialampHours;
+    double timeRemaining;
+    double ampsRemaining;
+    double rateofDepletion;
+    double hoursPrime;
+    double minutesPrime;
+    double secondsPrime;
+    double aveCurrentPrime;
+    double currentPrime = 0;
+    QTime t;
```

Comment:
`QTime t`, usually single letter variables are not meaningful enough, I would say that is true for this case as well, especially since you are using it as a private member and it there will be a lot of contexts where it won't have `QTime` beside it to show that it stands for time.

The major exception to this rule is in for loops, where `i` and `j` are very well known.
```c++
for(int i = 0; i < 5; i++)
{
    for(int j = 0; j < 4; i++)
    {
        // do something
    }
}
```